snippets/DispatchLoaderDefault.hpp

```diff
@@ -40,7 +40,7 @@
       }
   extern VULKAN_HPP_STORAGE_API DispatchLoaderDynamic defaultDispatchLoaderDynamic;
 #  else
-  static inline ::VULKAN_HPP_NAMESPACE::DispatchLoaderStatic & getDispatchLoaderStatic()
+  inline ::VULKAN_HPP_NAMESPACE::DispatchLoaderStatic & getDispatchLoaderStatic()
```

Comment:
In a follow-up PR, the module could also include `VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE`

Reply:
yes, the `VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE` has to be defined by the user because putting it into the header would violate the one-definition rule. But for C++ modules, we actually have a compiled C++ file that could hold that single definition. We're no longer header-only.

The dispatchers are function tables storing loaded functions from the drivers. When you don't want to link to `Vulkan.dll`, but instead dynamically load the instance level functions you need some place to store the function pointers. Vulkan-Hpp allows you to choose whether you want to link to Vulkan.so or whether to resolve the necessary functions during runtime.

Reply:
> But it would make a difference to the non-module handling, so I'm not so sure about it.

I think modules fundamentally change the equation: simply using `import vulkan;` will require consumers to have a *very* recent compiler and will enable most of the modern features that aren't really available in older compilers, so I feel it is OK to break with the non-module handling. I can add it in.

Reply:
I'd just like to clarify if there are any include guards, or we can just do:

```C++
module;

#include <vulkan/vulkan.hpp>
#include <vulkan/vulkan_raii.hpp>

export module vulkan;

VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE

export namespace VULKAN_HPP_NAMESPACE
{
...
```